src/script/mod.rs

```diff
@@ -268,9 +269,10 @@ impl Script {
         let interpreter_is_nushell = interpreter == "nushell" || interpreter == "nu";
 
         // Determine the valid script extensions based on the available interpreters.
-        let mut valid_script_extensions = Vec::new();
+        let mut valid_script_extensions = vec!["py", "pl", "r"];
```

Comment:
Since we are planning to remove the "build.sh" detection in the future, I think we should not add any additional file extensions here.